server/api/src/rsPhyPathReg.cpp

```diff
@@ -727,7 +727,32 @@ dirPathReg( rsComm_t *rsComm, dataObjInp_t *phyPathRegInp, char *filePath,
         fileStatInp_t fileStatInp;
         memset( &fileStatInp, 0, sizeof( fileStatInp ) );
 
-        snprintf( fileStatInp.fileName, MAX_NAME_LEN, "%s/%s", filePath, rodsDirent->d_name );
+        // Issue #3658 - https://github.com/irods/irods/issues/3658
+        //
+        // TODO:  This is a localized fix that addresses any trailing slashes
+        // in inPath. Any future refactoring of the code that deals with paths
+        // would most probably take care of this issue, and the code segment
+        // below would/should then be removed.
+        //
+        // This is the second part of a two part fix.  This section removes trailing
```

Comment:
reduce comment to just:
`This section removes trailing slashes from the directory path in the server when the path is already in the catalog.`